rotkehlchen/app_upgrades/manager.py

```diff
@@ -0,0 +1,49 @@
+import logging
+from typing import TYPE_CHECKING, Callable, Dict, Any, NamedTuple, Optional
+
+from rotkehlchen.app_upgrades.upgrades.upgrade_1 import app_upgrade_1
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+
+if TYPE_CHECKING:
+    from rotkehlchen.rotkehlchen import Rotkehlchen
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class UpgradeRecord(NamedTuple):
+    version: int
+    function: Callable
+    kwargs: Optional[Dict[str, Any]] = None
+
+
+UPGRADE_LIST = [
+    UpgradeRecord(version=1, function=app_upgrade_1),
+]
+
+
+class RotkiUpgradeManager:
+
+    def __init__(self, rotki: 'Rotkehlchen'):
+        self.rotki = rotki
+
+    def run_upgrades(self) -> None:
+        # get last upgrade setting
+        settings = self.rotki.data.db.get_settings()
+        current_upgrade = settings.last_app_upgrade
+        for upgrade in UPGRADE_LIST:
+            if current_upgrade < upgrade.version:
+                self._perform_upgrade(upgrade)
+                current_upgrade += 1
+        self.rotki.data.db.conn.cursor().execute(
```

Comment:
No since it counts the migrations that were executed since last one. Right now the only migration doesn't raise any error but if any migration could do it then this counter would be stored and there will be missing steps. I'm not sure right now about this